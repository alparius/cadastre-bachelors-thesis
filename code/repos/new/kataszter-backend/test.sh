#!/bin/bash

rm -f coverage.txt
echo 'mode: atomic' > coverage.txt

for d in $(go list ./... | grep -v testing); do
    go test -v -p 1 -race -covermode=atomic -coverprofile=coverage.out -coverpkg=$(go list ./...| grep -v testing | tr "\n" ",") $d
    if [ -f coverage.out ]; then
        tail -n +2 coverage.out >> coverage.txt || exit 255
        rm coverage.out
    fi
done
