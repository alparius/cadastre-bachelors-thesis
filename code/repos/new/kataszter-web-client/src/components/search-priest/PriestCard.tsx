import React from "react";
import { FormattedMessage as FM, useIntl } from "react-intl";
import { NavLink } from "react-router-dom";
import { Card, Header, Icon, Popup } from "semantic-ui-react";

import IPriestMin from "../../data/priest/PriestMin";
import ItalicFM from "../../util/styledMessages/ItalicFM";
import DateDisplay from "../DateDisplay";
import PriestCardPlacementList from "./PriestCardPlacementList";

interface IPriestCardProps {
    priest: IPriestMin;
}

const PriestCard: React.FC<IPriestCardProps> = (props: IPriestCardProps) => {
    const { priest } = props;
    const { formatMessage: fm } = useIntl();

    return (
        <Card key={priest.ID} fluid as={NavLink} to={`/priests/profile/${priest.ID}`}>
            <Card.Content>
                <Card.Header>
                    <Header as="h2">
                        {priest.name}
                        <span style={{ float: "right" }}>
                            {priest.verified ? (
                                <Popup
                                    content={fm({ id: "profile.verified.isVerifiedPopup" })}
                                    trigger={<Icon name="check circle outline" fitted color="green" />}
                                />
                            ) : (
                                <Popup
                                    content={fm({ id: "profile.verified.notVerifiedPopup" })}
                                    trigger={<Icon name="question circle outline" fitted color="grey" />}
                                />
                            )}
                        </span>
                    </Header>
                </Card.Header>
                <Card.Meta>
                    <FM id="label.born" />: {priest.birthTown ? priest.birthTown : <ItalicFM id="general.noData" />},{" "}
                    <DateDisplay date={priest.birthDate} /> - <FM id="label.died" />: <DateDisplay date={priest.deathDate} />
                </Card.Meta>
                <Card.Meta>
                    {priest.subscriptionDate.year && (
                        <>
                            <FM id="label.subscribed" />: <DateDisplay date={priest.subscriptionDate} />, <FM id="general.and" />{" "}
                        </>
                    )}
                    <FM id="label.graduated" />: <DateDisplay date={priest.graduationDate} />
                </Card.Meta>
                <Card.Meta>
                    <FM id="label.country" />: {priest.birthCountry ? priest.birthCountry : <ItalicFM id="general.noData" />}
                    {", "}
                    <FM id="label.county" />: {priest.birthCounty ? priest.birthCounty : <ItalicFM id="general.noData" />}
                    {", "}
                    <FM id="label.diocese" />: {priest.diocese ? priest.diocese : <ItalicFM id="general.noData" />}
                </Card.Meta>
                <Card.Description style={{ marginTop: "1em" }}>
                    <PriestCardPlacementList places={priest.assistantPriestPlaces} headerKey="label.priest.assistantPlacements" />
                    <PriestCardPlacementList places={priest.mainPriestPlaces} headerKey="label.priest.mainPlacements" />
                </Card.Description>
            </Card.Content>
        </Card>
    );
};

export default PriestCard;
