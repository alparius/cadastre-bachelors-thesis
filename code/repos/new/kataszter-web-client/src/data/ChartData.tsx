interface IChartData {
    key: any;
    data: any;
    counter: number;
}

export default IChartData;
