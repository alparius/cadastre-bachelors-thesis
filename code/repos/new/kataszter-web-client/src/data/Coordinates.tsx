interface ICoordinates {
    lat: number;
    lng: number;
}

export let NewCoordinates: ICoordinates = {
    lat: 0,
    lng: 0
};

export default ICoordinates;
