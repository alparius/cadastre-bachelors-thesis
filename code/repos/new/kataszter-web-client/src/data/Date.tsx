interface IDate {
    year: number;
    month: number;
    day: number;
    comment: string;
}

export let NewDate: IDate = {
    year: 0,
    month: 0,
    day: 0,
    comment: ""
};

export default IDate;
