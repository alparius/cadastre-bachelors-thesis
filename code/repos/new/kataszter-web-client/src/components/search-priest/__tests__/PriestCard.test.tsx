/*===================
  PriestCard.unit.test.tsx
 ===================*/
import React from "react";

import IPriestMin from "../../../data/priest/PriestMin";
import { NewTestDate, NewTestPriestMin } from "../../../util/test/data/test-data";
import { createComponentWithIntlAndRouter } from "../../../util/test/test-wrapper";
import PriestCard from "../PriestCard";

describe("PriestCard element", () => {
    it("should render a card with the priest data", () => {
        const regularPriest: IPriestMin = NewTestPriestMin;
        const container = createComponentWithIntlAndRouter(<PriestCard priest={regularPriest} />).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render a card with with the priest data and 'no data' labels for missing fields", () => {
        const priestWithMissingFields: IPriestMin = {
            id: 1,
            name: "name",
            birthDate: NewTestDate,
            deathDate: NewTestDate,
            graduationDate: NewTestDate,
            retirementDate: NewTestDate,
            subscriptionDate: NewTestDate
        } as any;
        const container = createComponentWithIntlAndRouter(<PriestCard priest={priestWithMissingFields} />).toJSON();
        expect(container).toMatchSnapshot();
    });
});
