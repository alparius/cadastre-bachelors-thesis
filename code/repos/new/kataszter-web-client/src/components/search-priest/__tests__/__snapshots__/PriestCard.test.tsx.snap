// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriestCard element should render a card with the priest data 1`] = `
<a
  aria-current={null}
  className="ui fluid card"
  href="/priests/profile/1"
  onClick={[Function]}
>
  <div
    className="content"
  >
    <div
      className="header"
    >
      <h2
        className="ui header"
      >
        name
        <span
          style={
            Object {
              "float": "right",
            }
          }
        >
          <i
            aria-hidden="true"
            className="grey question circle outline fitted icon"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          />
        </span>
      </h2>
    </div>
    <div
      className="meta"
    >
      born
      : 
      birthTown
      ,
       
      <span
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        2000.06.06
      </span>
       - 
      died
      : 
      <span
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        2000.06.06
      </span>
    </div>
    <div
      className="meta"
    >
      subscribed
      : 
      <span
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        2000.06.06
      </span>
      , 
      and
       
      graduated
      : 
      <span
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        2000.06.06
      </span>
    </div>
    <div
      className="meta"
    >
      country
      : 
      birthCountry
      , 
      county
      : 
      birthCounty
      , 
      diocese
      : 
      diocese
    </div>
    <div
      className="description"
      style={
        Object {
          "marginTop": "1em",
        }
      }
    >
      <div
        className="ui list"
        role="list"
      >
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="header"
          >
            Assistant priest placements
          </div>
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <a
            aria-current={null}
            className="hoverLink"
            href="/parishes/profile/0"
            onClick={[Function]}
          >
             
            place
             
          </a>
          : 
          <span
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            2000.06.06
          </span>
           - 
          <span
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            2000.06.06
          </span>
        </div>
      </div>
      <div
        className="ui list"
        role="list"
      >
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="header"
          >
            Main priest placements
          </div>
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <a
            aria-current={null}
            className="hoverLink"
            href="/parishes/profile/0"
            onClick={[Function]}
          >
             
            place
             
          </a>
          : 
          <span
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            2000.06.06
          </span>
           - 
          <span
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            2000.06.06
          </span>
        </div>
      </div>
    </div>
  </div>
</a>
`;

exports[`PriestCard element should render a card with with the priest data and 'no data' labels for missing fields 1`] = `
<a
  aria-current={null}
  className="ui fluid card"
  href="/priests/profile/undefined"
  onClick={[Function]}
>
  <div
    className="content"
  >
    <div
      className="header"
    >
      <h2
        className="ui header"
      >
        name
        <span
          style={
            Object {
              "float": "right",
            }
          }
        >
          <i
            aria-hidden="true"
            className="grey question circle outline fitted icon"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          />
        </span>
      </h2>
    </div>
    <div
      className="meta"
    >
      born
      : 
      <i>
        No data
      </i>
      ,
       
      <span
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        2000.06.06
      </span>
       - 
      died
      : 
      <span
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        2000.06.06
      </span>
    </div>
    <div
      className="meta"
    >
      subscribed
      : 
      <span
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        2000.06.06
      </span>
      , 
      and
       
      graduated
      : 
      <span
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        2000.06.06
      </span>
    </div>
    <div
      className="meta"
    >
      country
      : 
      <i>
        No data
      </i>
      , 
      county
      : 
      <i>
        No data
      </i>
      , 
      diocese
      : 
      <i>
        No data
      </i>
    </div>
    <div
      className="description"
      style={
        Object {
          "marginTop": "1em",
        }
      }
    >
      <div
        className="ui list"
        role="list"
      >
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="header"
          >
            Assistant priest placements
          </div>
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <i>
            No data
          </i>
        </div>
      </div>
      <div
        className="ui list"
        role="list"
      >
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="header"
          >
            Main priest placements
          </div>
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <i>
            No data
          </i>
        </div>
      </div>
    </div>
  </div>
</a>
`;
